python/tvm/runtime/object_path.py

```diff
@@ -122,3 +123,14 @@ class MapValuePath(ObjectPath):
 @tvm._ffi.register_object("MissingMapEntryPath")
 class MissingMapEntryPath(ObjectPath):
     pass
+
+
+@tvm._ffi.register_object("ObjectPathPair")
+class ObjectPathPair(Object):
+    @property
+    def lhs_path(self) -> ObjectPath:
+        return _ffi_node_api.ObjectPathPairLhsPath(self)
+
+    @property
+    def rhs_path(self) -> ObjectPath:
+        return _ffi_node_api.ObjectPathPairRhsPath(self)
```

Comment:
If you reflection is defined inside `ObjectPathPair`, then we don't need any boilerplate code for attribute access, e.g.

```C++
// inside `ObjectPathPairNode`
void VisitAttrs(tvm::AttrVisitor* v) {
  v->Visit("lhs_path", &lhs_path);
  v->Visit("rhs_path", &rhs_path);
}
// in a cc file

- TVM_REGISTER_OBJECT_TYPE(ObjectPathPairNode);
+ TVM_REGISTER_NODE_TYPE(ObjectPathPairNode);
```

Reply:
The problem with this approach is that we can't `#include "reflection.h"` in the `structural_equal.h" because of a circular dependency (`reflection.h` already includes `structural_equal.h`). We could forward-declare the `AttrVisitor` class in the header, and put the implementation of `VisitAttrs` in the cc file, but we would risk getting the two out of sync. I think this is really a special case because this ObjectPathPair is part of the reflection mechanism, so I think it is okay to have the extra boilerplate for the getters.

Reply:
Ah I see. It is definitely less convenient but fortunately it's the only case :-)